[Workers] Restructure & update logging section

markdembo opened the pull request:
* Restructure docs for logging

```
Old:
 Observability
  ...
  Log from Workers
  Log Push
  Tail Workers
  
  
New:
 Observability
  ...
  Logging
     Real-time logs
     Log Push
     Tail Workers
 ```
 * Update log persisting methods
 * Call out limits separately
  * Add information on real-time log filtering
 * Remove old, slightly outdated, Youtube video
 * Add redirects
 * Fix all broken references to logging docs

tanushreesharma-cf commented on content/workers/observability/logging/dashboard.md:
```diff
@@ -0,0 +1,105 @@
+  description: Debug your Worker application by accessing logs and exceptions through the Cloudflare dashboard.
+---
+
+# Log from Workers
```
```suggestion
# Real-time logs in the Dashboard 
```

tanushreesharma-cf commented on content/workers/observability/logging/dashboard.md:
```diff
@@ -0,0 +1,105 @@
+Note that:
+
+- Workers logs are not stored. You can start and stop the stream at any time to view them, but they do not persist.
+- Logs will not display if the Worker's requests per second are over 200 for the last 5 minutes.
```
@admah is this still true? I know we did some scaling work a couple quarters ago.

tanushreesharma-cf commented on content/workers/observability/logging/wranger-tail.md:
```diff
@@ -1,17 +1,15 @@
+  description: Debug your Worker application by accessing logs and exceptions through `wrangler tail`.
 ---
 
 # Log from Workers
```
```suggestion
# Real-time logs with wrangler tail
```

tanushreesharma-cf commented on content/workers/observability/logging/dashboard.md:
```diff
@@ -0,0 +1,105 @@
+# Log from Workers
+
+Debugging is a critical part of developing a new application — whether running code in the initial stages of development, or trying to understand an issue occurring in production.
+
```
```suggestion
You can access logs and exceptions for your Workers by logging into [the dashboard](/workers/observability/log-from-workers/#view-logs-from-the-dashboard). 
```

markdembo commented:
Addressed feedback from Tanushree & Kate.
Added redirect & fixed all broken links throughout the documentation.
Waiting for feedback from @admah on one pending detail.

deadlypants1973 commented on content/workers/observability/logging/real-time-logs.md:
```diff
@@ -9,10 +9,6 @@ meta:
-
 The Workers platform captures all `console.log`'s and uncaught exceptions, in addition to information about the event itself.
 
 {{<Aside type="warning">}}
```
What feature are we talking about here? Logging in general?

markdembo replied:
Clarifying with @tanushreesharma-cf and @admah. Will specify.

deadlypants1973 commented on content/workers/observability/logging/real-time-logs.md:
```diff
@@ -21,15 +17,10 @@ This feature is not available for zones on the [Cloudflare China Network](/china
 ## Add custom logs
 
-Any `console.log` statements within your Worker will appear within `wrangler tail` and the dashboard output. The following example demonstrates a custom `console.log` within a Worker request handler.
+Any `console.log` statements within your Worker will be visible in the dashboard or wrangler tail. The following example demonstrates a custom `console.log` within a Worker request handler.
```
```suggestion
Any `console.log` statements within your Worker will be visible in the dashboard or [`wrangler tail`](/workers/wrangler/commands/#tail). The following example demonstrates a custom `console.log` within a Worker request handler.
```

deadlypants1973 commented on content/workers/observability/logging/real-time-logs.md:
```diff
@@ -9,10 +9,6 @@ meta:
 
 Debugging is a critical part of developing a new application — whether running code in the initial stages of development, or trying to understand an issue occurring in production.
```
@markdembo Can we rewrite this to talk more about what's going on in this page? This mentions "debugging" which is a concept that could apply to the entire Observability section and seems like a relic of this page's old identity, but this page is titled "real-time logs". We should write something about real-time logs and address that this page goes into 2 methods for that process: dashboard or `wrangler tail`.

deadlypants1973 commented on content/workers/observability/logging/real-time-logs.md:
```diff
@@ -21,15 +17,10 @@ This feature is not available for zones on the [Cloudflare China Network](/china
-
-{{</Aside>}}
 
 ## Add custom logs
```
Is this step a pre-cursor to the next two H2's that talk about viewing logs from dash or Wrangler? We could make this clearer - otherwise they all look like separate concepts. 

Would this code go into an `index.js` or any file in the Worker project?

deadlypants1973 commented on content/workers/observability/logging/real-time-logs.md:
```diff
@@ -73,21 +64,25 @@ async function handleRequest(request) {
 {{</tabs>}}
 
-After you deploy the above code, run `wrangler tail` in your terminal, and then access your Worker. Your terminal will display:
+After you deploy the above code you can see the real-time logs in the dashboard or `wrangler tail`.
```
```suggestion
After you deploy the [above code](/workers/observability/logging/real-time-logs/#add-custom-logs), view the real-time logs in the dashboard or `wrangler tail`.
```

deadlypants1973 commented on content/workers/observability/logging/real-time-logs.md:
```diff
@@ -73,21 +64,25 @@ async function handleRequest(request) {
+## View logs using `wrangler tail`
 
-## Use `wrangler tail`
+To view real-time logs associated with any deployed Worker using wrangler:
```
```suggestion
To view real-time logs associated with any deployed Worker using Wrangler:
```

deadlypants1973 commented on content/workers/observability/logging/real-time-logs.md:
```diff
@@ -73,21 +64,25 @@ async function handleRequest(request) {
+To view real-time logs associated with any deployed Worker using wrangler:
 
-With your Workers application deployed, you may want to inspect incoming traffic. This may be useful in situations where a user is running into production issues that they cannot easily reproduce. In these instances, [`wrangler tail`](/workers/wrangler/commands/#tail) allows developers to livestream their Workers application’s logs, giving real-time insight into their application's incoming reque ...
+1. Navigate to your Worker project directory
+2. Run `npx wrangler tail`
```
```suggestion
1. Go to your Worker project directory.
2. Run `npx wrangler tail`.
```

deadlypants1973 commented on content/workers/observability/logging/real-time-logs.md:
```diff
@@ -120,31 +115,37 @@ $ npx wrangler tail | jq .event.request.url
 
 - Workers logs are not stored. You can start and stop the stream at any time to view them, but they do not persist.
-- Logs will not display if the Worker's requests per second are over 200 for the last 5 minutes.
+- If your Worker has a high volume of traffic, the real-time logs might enter sampling mode. This will cause some of your messages to be dropped and a warning to appear in your logs. 
```
```suggestion
- If your Worker has a high volume of traffic, the real-time logs might enter sampling mode. This will cause some of your messages to be dropped and a warning to appear in your logs.
```

deadlypants1973 commented on content/workers/observability/logging/real-time-logs.md:
```diff
@@ -120,31 +115,37 @@ $ npx wrangler tail | jq .event.request.url
-## Push logs to storage
+{{<Aside type="note">}}
+
+You can filter real-time logs both in the dashboard and using [wrangler tail](/workers/wrangler/commands/#tail). This can help mitgate messages from being dropped, if your Worker has a high volume of messages.
```
```suggestion
You can filter real-time logs in the dashboard or using [`wrangler tail`](/workers/wrangler/commands/#tail). If your Worker has a high volume of messages, filtering real-time logs can help mitgate messages from being dropped.
```

deadlypants1973 commented on content/workers/observability/logging/real-time-logs.md:
```diff
@@ -120,31 +115,37 @@ $ npx wrangler tail | jq .event.request.url
+
+{{</Aside>}}
+
+## Persisting logs
```
```suggestion
## Persist logs
```

deadlypants1973 commented on content/workers/observability/logging/real-time-logs.md:
```diff
@@ -120,31 +115,37 @@ $ npx wrangler tail | jq .event.request.url
+
+## Persisting logs
+
+Logs can be persited in two ways: using [Workers Logpush](/workers/observability/logging/logpush/) or [Tail Workers](/workers/observability/logging/tail-workers/).
```
```suggestion
Logs can be persited in two ways:

1. [Workers Logpush](/workers/observability/logging/logpush/).
2. [Tail Workers](/workers/observability/logging/tail-workers/).
```

deadlypants1973 reviewed (commented):
PCX

deadlypants1973 commented on content/workers/observability/logging/real-time-logs.md:
```diff
@@ -73,21 +66,25 @@ async function handleRequest(request) {
 {{</tabs>}}
 
-After you deploy the above code, run `wrangler tail` in your terminal, and then access your Worker. Your terminal will display:
+After you deploy the [above code](/workers/observability/logging/real-time-logs/#add-custom-logs), view the real-time logs in the dashboard or `wrangler tail`.
```
```suggestion
After you deploy the [above code](/workers/observability/logging/real-time-logs/#add-custom-logs), view the real-time logs in [the dashboard](/workers/observability/logging/real-time-logs/#view-logs-from-the-dashboard) or [`wrangler tail`](/workers/observability/logging/real-time-logs/#view-logs-using-wrangler-tail).
```

deadlypants1973 commented on content/workers/observability/logging/real-time-logs.md:
```diff
@@ -73,21 +66,25 @@ async function handleRequest(request) {
 
-With your Workers application deployed, you may want to inspect incoming traffic. This may be useful in situations where a user is running into production issues that they cannot easily reproduce. In these instances, [`wrangler tail`](/workers/wrangler/commands/#tail) allows developers to livestream their Workers application’s logs, giving real-time insight into their application's incoming reque ...
+1. Go to your Worker project directory.
+2. Run `npx wrangler tail`.
```
```suggestion
2. Run [`npx wrangler tail`](/workers/wrangler/commands/#tail).
```

deadlypants1973 commented on content/workers/observability/logging/real-time-logs.md:
```diff
@@ -1,35 +1,28 @@
-title: Log from Workers
+title: Real-time logs
 meta:
   description: Debug your Worker application by accessing logs and exceptions through the Cloudflare dashboard or `wrangler tail`.
```
```suggestion
  description: View real-time logs of your Worker application through the Cloudflare dashboard or `wrangler tail`.
```
